tests/hardhat/Liquidator/restrictedLiquidations.ts

```diff
@@ -0,0 +1,196 @@
+import { FakeContract, MockContract, smock } from "@defi-wonderland/smock";
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import chai from "chai";
+import { Signer, constants } from "ethers";
+import { ethers } from "hardhat";
+
+import { convertToUnit } from "../../../helpers/utils";
+import {
+  Comptroller,
+  Liquidator__factory,
+  Liquidator,
+  VBNB,
+  VBep20Immutable,
+  VAIController
+} from "../../../typechain";
+import { ComptrollerErrorReporter } from "../util/Errors";
+
+const { expect } = chai;
+chai.use(smock.matchers);
+
+type LiquidatorFixture = {
+  vaiController: FakeContract<VAIController>;
+  vBep20: FakeContract<VBep20Immutable>;
+  vBnb: FakeContract<VBNB>;
+  comptroller: FakeContract<Comptroller>;
+  liquidator: MockContract<Liquidator>;
+};
+
+async function deployLiquidator(): Promise<LiquidatorFixture> {
+  const [admin, treasury] = await ethers.getSigners();
+  const treasuryPercentMantissa = convertToUnit("0.05", 18);
+
+  const comptroller = await smock.fake<Comptroller>("Comptroller");
+  const vaiController = await smock.fake<VAIController>("VAIController");
+  const vBnb = await smock.fake<VBNB>("VBNB");
+  const vBep20 = await smock.fake<VBep20Immutable>("VBep20Immutable");
+
+  const Liquidator = await smock.mock<Liquidator__factory>("Liquidator");
+  const liquidator = await Liquidator.deploy(
+    admin.address,
+    vBnb.address,
+    comptroller.address,
+    vaiController.address,
+    treasury.address,
+    treasuryPercentMantissa
+  );
+  
+  return { comptroller, vaiController, vBnb, vBep20, liquidator };
+}
+
+function configure(fixture: LiquidatorFixture) {
+  const { comptroller } = fixture;
+  comptroller.liquidationIncentiveMantissa.returns(convertToUnit("1.1", 18));
+}
+
+describe("Liquidator", () => {
+  let admin: SignerWithAddress;
+  let treasury: SignerWithAddress;
+  let borrower: SignerWithAddress;
+  let guy: SignerWithAddress;
+  let accounts: SignerWithAddress[];
+
+  let vaiController: FakeContract<VAIController>;
+  let vBep20: FakeContract<VBep20Immutable>;
+  let vBnb: FakeContract<VBNB>;
+  let comptroller: FakeContract<Comptroller>;
+  let liquidator: MockContract<Liquidator>;
+
+  beforeEach(async () => {
+    [admin, treasury, borrower, guy, ...accounts] = await ethers.getSigners();
```

Comment:
is `guy` account the liquidator one?

Reply:
Depends on the test, it maybe "someone pretending to be an admin". I just don't want to have a million of accounts in the test suite